src/parser/nmodl_driver.cpp

```diff
@@ -24,60 +24,107 @@ std::shared_ptr<ast::Program> NmodlDriver::parse_stream(std::istream& in) {
     NmodlLexer scanner(*this, &in);
     NmodlParser parser(scanner, *this);
 
-    this->lexer = &scanner;
-    this->parser = &parser;
-
     scanner.set_debug(trace_scanner);
     parser.set_debug_level(trace_parser);
     parser.parse();
     return astRoot;
 }
 
-//// parse nmodl file
-std::shared_ptr<ast::Program> NmodlDriver::parse_file(const std::string& filename) {
+std::shared_ptr<ast::Program> NmodlDriver::parse_file(const std::string& filename,
+                                                      const location* loc) {
     std::ifstream in(filename.c_str());
     stream_name = filename;
 
     if (!in.good()) {
-        logger->error("Can not open file : {}", filename);
-        return nullptr;
+        std::ostringstream oss;
+        if (loc == nullptr) {
+            oss << "NMODL Parser Error : ";
+        }
+        oss << "can not open file : " << filename;
+        if (loc != nullptr) {
+            parse_error(*loc, oss.str());
+        } else {
+            throw std::runtime_error(oss.str());
+        }
     }
     parse_stream(in);
     return astRoot;
 }
 
-/// parser nmodl provided as string (used for testing)
 std::shared_ptr<ast::Program> NmodlDriver::parse_string(const std::string& input) {
     std::istringstream iss(input);
     parse_stream(iss);
     return astRoot;
 }
 
-void NmodlDriver::error(const std::string& m, const class location& l) {
-    std::cerr << l << " : " << m << std::endl;
-}
+std::shared_ptr<ast::Include> NmodlDriver::parse_include(const std::string& name,
+                                                         const location& loc) {
+    // Try to find directory containing the file to import
+    const auto directory_path = library.find_file(name);
+
+    // Complete path of file (directory + filename).
+    std::string absolute_path = name;
+
+    if (!directory_path.empty()) {
+        absolute_path = directory_path + std::string(1, utils::pathsep) + name;
```

Comment:
Here the `directory_path` I believe should have the path of the `mod` file which included the `mod` file. I need to check also how `mod2c` takes care of this case

Reply:
I think the issue was FileLibrary : it was missing the directory of the top parsed file in the list of directories to look for .mod files

But now I have another issue:

```
bash-3.2$ bin/nmodl ../test/integration/mod/cabpump.mod
[NMODL] [info] :: Processing ../test/integration/mod/cabpump.mod
libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: NMODL Parser Error : can not open file : var_init.inc [Location : 43.9-22]
Abort trap: 6
bash-3.2$ bin/nmodl ../test/integration/mod/cabpump.mod
[NMODL] [info] :: Processing ../test/integration/mod/cabpump.mod
[NMODL] [info] :: Running symtab visitor
[NMODL] [info] :: Running LOCAL to ASSIGNED visitor
[NMODL] [info] :: Running code compatibility checker
[NMODL] [info] :: Running verbatim rename visitor
[NMODL] [info] :: Running KINETIC block visitor
[NMODL] [info] :: Running STEADYSTATE visitor
[NMODL] [info] :: Parsing Units
[NMODL] [info] :: Running cnexp visitor
Assertion failed: (solve_node_symbol != nullptr), function create_solution_expression, file /Users/tristan/src/github.com/BlueBrain/nmodl/src/visitors/solve_block_visitor.cpp, line 45.
Abort trap: 6
```